dataland-qa-service/src/main/resources/application-nodb.properties

```diff
@@ -0,0 +1,8 @@
+spring.security.oauth2.resourceserver.jwt.issuer-uri=https://local-dev.dataland.com/keycloak/realms/datalandsecurity
+spring.security.oauth2.resourceserver.jwt.jwk-set-uri=https://local-dev.dataland.com/keycloak/realms/datalandsecurity/protocol/openid-connect/certs
+spring.datasource.driver-class-name=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=password
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.rabbitmq.listener.simple.auto-startup=false
```

Comment:
The qa-service uses the keycloak adapter, and the keycloak adapter uses the application prop `@Value("\${dataland.apikeymanager.base-url:}") private val apiKeyManagerBaseUrl: String,` in `WebSecurityConfig.kt`.

So I wonder how the qa-service can start without that prop.

Same goes for the `application.properties` file


During the application startup process, Spring attempts to resolve all the @Value annotations and inject the corresponding property values.

Reply:
We don't have api-key support enabled for the qa manager because I don't see use for it. Except reviewers want to create their own frontend :). The remaining reasoning is the same as you did for `dataland.internalstorage.base-url`.

Reply:
Yes but anyways the startup would usually fail because @Value cannot find the property.

I think the reason it does not is the ":" at the end of `dataland.apikeymanager.base-url`.
It tells spring to resolve the value to an empty string if the value cannot be found in the `application.properties`.

Reply:
I resolve this because I have understood this now.